### ISSUE TITLE
v0.3.5 breaks the public API

### ISSUE THREAD

**Issue by iliana** [wilsonzlin/ecma-rs#5]
Hi, just wanted to report a build failure regression from v0.3.4 to v0.3.5. I noticed there are a number of releases with new minor versions over the past day, but I would recommend yanking the v0.3.5 release, as using `cargo update` will pull in a breaking change which affects code using minify-js v0.2 (such as minify-html v0.10).

https://github.com/wilsonzlin/parse-js/commit/511899590ed7e00139f7333ea479795017735192 changes inner types on public enums and public function return types.

**Comment by wilsonzlin** [comment]
Thanks for raising, yes this version should have been a minor (instead of patch) bump, I've yanked `0.3.5`.

Version `0.3.2` also introduced a new enum variant to support `import.meta`, and `0.3.3` and `0.3.4` introduced new enum variants to support JSX, so I've yanked those versions as well. I just had a look at the [official Cargo SemVer guidelines](https://doc.rust-lang.org/cargo/reference/semver.html#enum-variant-new) and now realise that new enum variants count as a major change unless explicitly tagged with `non_exhaustive`, so will take this into consideration for future releases.
